### PR TITLE
feat(js): add SWC support

### PR DESCRIPTION
**Pull request by nartc**
## Current Behavior

No general SWC support

## Expected Behavior

This PR adds SWC support to `nrwl/js` plugin which the other plugins can make use of

### CODE REVIEW

**Review by AgentEnder — CHANGES_REQUESTED** [review]

File: packages/js/src/executors/swc/swc.impl.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  //     context.root,
+  //     context.projectName,
+  //     context.
+  //     targetName,
+  //     context.configurationName,
+  //     target,
+  //     dependencies,
+  //     options.buildableProjectDepsInPackageJsonType
+  //   );
+  // }
```
[the remark this answers is not in the document]
Comment (nartc) [reply packages/js/src/executors/swc/swc.impl.ts:123]: We might need it. Depends on how Victor's porting the old implementation over.

[the remark this answers is not in the document]
Comment (nartc) [reply packages/js/src/executors/swc/swc.impl.ts]: removed

File: packages/js/src/utils/swc/compile-swc.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import { exec, execSync } from 'child_process';
+import { normalizeTsCompilationOptions } from '../normalize-ts-compilation-options';
+
+export async function compileSwc(
+  tsCompilationOptions: TypeScriptCompilationOptions,
+  postCompilationCallback: () => void | Promise<void>
+): Promise<{ success: boolean }> {
+  const normalizedOptions = normalizeTsCompilationOptions(tsCompilationOptions);
+
+  console.log(`Compiling with SWC for ${normalizedOptions.projectName}...`);
```
Comment (jaysoo) [inline packages/js/src/utils/swc/compile-swc.ts:11]: Is this for debugging, or do we want to log? Use `logger` from `@nrwl/devkit` if it's the latter.

File: packages/js/src/utils/swc/compile-swc.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  // Open issue: https://github.com/swc-project/cli/issues/20
+  let swcCmd = `npx swc ${srcPath} -d ${destPath} --source-maps --config-file=${swcrcPath}`;
+
+  if (normalizedOptions.watch) {
+    swcCmd += ' --watch';
+    return createSwcWatchProcess(swcCmd, postCompilationCallback);
+  }
+
+  const swcCmdLog = execSync(swcCmd).toString();
+  console.log(swcCmdLog.replace(/\n/, ''));
```
Comment (jaysoo) [inline packages/js/src/utils/swc/compile-swc.ts]: Remove this?

File: packages/js/src/utils/swc/compile-swc.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    });
+
+    const processExitListener = () => watchProcess.kill();
+
+    process.on('SIGINT', processExitListener);
+    process.on('SIGTERM', processExitListener);
+    process.on('exit', processExitListener);
+
+    watchProcess.on('exit', (args) => {
+      console.log('exit', args);
```
Comment (jaysoo) [inline packages/js/src/utils/swc/compile-swc.ts]: Remove

File: packages/js/src/utils/swc/compile-swc.ts (hunk 2 above)
Comment (nartc) [reply packages/js/src/utils/swc/compile-swc.ts]: Fixed

**Review by jaysoo — CHANGES_REQUESTED** [review]

File: packages/js/src/utils/swc/compile-swc.ts (hunk 3 above)
Comment (nartc) [reply packages/js/src/utils/swc/compile-swc.ts]: This one is to log what `swcCmd` logs which is to show how many files were compiled by SWC.

File: packages/js/src/utils/swc/compile-swc.ts (hunk 4 above)
Comment (nartc) [reply packages/js/src/utils/swc/compile-swc.ts]: removed

**Review by AgentEnder — APPROVED** [review]
Looks good 🎉

**Pull request merged**